lib/public_activity.rb

```diff
@@ -40,6 +41,34 @@ def self.enabled?
     !!PublicActivity.config.enabled
   end
 
+  # Execute the code block with PublicActiviy active
+  #
+  # Example usage:
+  #   PublicActivity.with_tracking do
+  #     # your test code here
+  #   end
+  def self.with_tracking
+    current = PublicActivity.config.enabled
+    PublicActivity.config.enabled = true
+    yield
+  ensure
+    PublicActivity.config.enabled = current
+  end
+
```

Comment:
@sungwoncho This will make sure it goes back to what the value used to be before, so you can worry less and have clean semantics.

As for the  deprecation of `.enabled=false` it makes sense to disable p_a functionality per class or temporarily. Global turn-off switch is problematic and if you don't want p_a at all you just don't use it.